core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1624,6 +1626,24 @@ private RexNode simplifyOrs(List<RexNode> terms, RexUnknownAs unknownAs) {
             continue;
           }
         }
+        break;
+      case NOT_EQUALS:
+        Comparison notEqualsComparison =
+            Comparison.of(term, node -> RexUtil.isReferenceOrAccess(node, false));
+        if (notEqualsComparison != null) {
+          Comparison prevNotEqualsComparison = notEqualsComparisonMap.get(notEqualsComparison.ref);
+          if (prevNotEqualsComparison == null) {
+            notEqualsComparisonMap.put(notEqualsComparison.ref, notEqualsComparison);
+          } else {
+            Comparable comparable1 = notEqualsComparison.literal.getValue();
+            Comparable comparable2 = prevNotEqualsComparison.literal.getValue();
+            //noinspection unchecked
+            if (comparable1.compareTo(comparable2) != 0) {
+              return rexBuilder.makeLiteral(true);
```

Comment:
what if `x != 0 || x!= 1` in case x is a nullable variable?
when `x is null`; I think the expression should evaluate to `null` instead of `true`

I'm not sure if this is right now in a specific unknownas mode (doesn't seems so); or the operands are non nullables - but the best is to assume that operand of the comparision might be nullable without even considering the actual unknownAs mode.

the important thing is that I think at this point you may only conclude that `x is not null or null`
Which is `true` in some cases:
* like when `x is of a type which is not nullable`
* or in unknownas.TRUE mode

you may narrow down this to match only the known good ones; or use the above generalized fact and let other parts of reximplify to do the rest

Reply:
Thanks for your feedback, @kgyrtkirk .

If I understand correctly, you say that the simplification
`x != 0 || x!= 1 => TRUE`
Can only be done in unknownas.TRUE mode or if x's type is not nullable.

Regarding the expression:
`x = 0 && x = 1 => FALSE`
I have verified, and right now it seems that this is only simplified with unknownas.FALSE. I assume this behavior is correct, right?

Moreover, when you say that 
> you may only conclude that `x is not null or null`. Which is `true` in some cases
I think right now this simplification is performed with all unknownas modes (TRUE / FALSE / UNKNOWN), I have checked via this test:
```
checkSimplify(
    or(isNull(aRef),
        isNotNull(aRef)),
    "true");
```

Reply:
@kgyrtkirk when you have some time, please check the latest status of the PR, which should handle the nullability scenario that you mentioned.